packages/studio-ui/src/web/components/Layout/TopNav/cloud/local-storage-pat.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect } from 'react'
+import { fetchPatStatus } from './pat'
+
+const LOCALSTORAGE_KEY = 'bp/pat'
+
+export const LocalStoragePat = (props: { onCompleted: (value?: string) => void }): JSX.Element => {
+  const { onCompleted } = props
+
+  useEffect(() => {
+    const initialPatValue = localStorage.getItem(LOCALSTORAGE_KEY)
+
+    const ac = new AbortController()
```

Comment:
**question:** why use a cancellable request everywhere ? is this absolutely necessary ? can it lead to some non-desirable side-effects ? (e.g components gets unmounted but the user still wants chatbot to be deployed)

Reply:
It is needed to avoid memory leaks. If a promise is not settled when a component is unmounted, [React will complain](https://stackoverflow.com/questions/54954385/react-useeffect-causing-cant-perform-a-react-state-update-on-an-unmounted-comp).

Cancelling requests avoid these potential memory leaks